websites/Z/Zee5/presence.ts

```diff
@@ -0,0 +1,103 @@
+const presence = new Presence({
+		clientId: "1103408220921397349",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+/**
+ * Get Timestamps
+ * @param {Number} videoTime Current video time seconds
+ * @param {Number} videoDuration Video duration seconds
+ */
+function getTimestamps(videoTime: number, videoDuration: number): number[] {
+	const startTime = Date.now();
+	return [
+		Math.floor(startTime / 1000),
+		Math.floor(startTime / 1000) - videoTime + videoDuration,
+	];
+}
+
+const getVideoStatus = (
+	presenceData: PresenceData,
+	video: HTMLVideoElement
+) => {
+	if (video.paused) {
+		presenceData.smallImageKey = "paused";
+		presenceData.smallImageText = "paused";
+		delete presenceData.startTimestamp;
+	} else {
+		const [startTimestamp, endTimestamp] = getTimestamps(
+			Math.floor(video.currentTime),
+			Math.floor(video.duration)
+		);
+		presenceData.startTimestamp = startTimestamp;
+		presenceData.endTimestamp = endTimestamp;
+		presenceData.smallImageKey = "playing";
+		presenceData.smallImageText = "playing";
+	}
+	return presenceData;
+};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/H9fXid0.png",
+		startTimestamp: browsingTimestamp,
+	};
+
+	if (document.location.pathname === "/")
+		presenceData.details = "Browsing home page";
+	else if (document.location.pathname === "/movies")
+		presenceData.details = "Browsing  movies";
+	else if (document.location.pathname.includes("/movies/details")) {
+		const currentPresenceData = getVideoStatus(
+			presenceData,
+			document.querySelector<HTMLVideoElement>(
+				"div.playerContainer > div >video"
+			)
+		);
+		currentPresenceData.details = "Watching";
+		currentPresenceData.state = document.querySelector<HTMLElement>(
+			"div.movieDetailContainer > div > div > div > div.metaDataDiv > div.consumptionMetaDiv >  div  > h1 "
+		).textContent;
+	} else if (document.location.pathname.includes("tv-shows/details")) {
+		const currentPresenceData = getVideoStatus(
+			presenceData,
+			document.querySelector<HTMLVideoElement>(
+				"div.playerContainer > div >video"
+			)
+		);
+		currentPresenceData.details = "Watching";
+		currentPresenceData.state = document.querySelector<HTMLElement>(
+			"div.episodeDetailContainer > div > div > div > div.metaDataDiv > div.consumptionMetaDiv >  div  > h1 "
+		).textContent;
```

Comment:
Let's shorten selectors. 
I used this [series link](https://www.zee5.com/global/tv-shows/details/bhagya-lakshmi/0-6-3392)
E.g.
```suggestion
		currentPresenceData.state = document.querySelector<HTMLElement>(
			'[class="title"]'
		).textContent;
```

Explanation:

Example 1:
```html
<h1 class="title">Bhagya Lakshmi</h1>
```
Whenever the class name is only used once as in here with title we can queryselector it by using:
```ts
// no reason to use a type if u don't access special attributes nor reassign it.
document.querySelector(
			'[class="title"]'
		)
```

Example 2:
```html
<meta property="og:title" content="Watch Bhagya Lakshmi Latest Episodes Online Exclusively on ZEE5" data-react-helmet="true">
```

this meta property is only used once. So we can use the following code:
```ts
// Selector => ['meta[property="og:title"] 
// a meta element with the property og:title

// there's a reason to use type if u want to do it this way:
document.querySelector<HTMLMetaElement>('meta[property="og:title"]').content

// but also not cause you could do this:
document.querySelector('meta[property="og:title"]').getAttribute("content")
```

Reply:
we can't use the metadata title as it contains long phrase. we can use the above suggested changes with title class

Reply:
seems like the h1 with class only available on movie homepage when we play that episode we don't have the title class. so we can't use the title tag so shorten this we can get the title by the following
`document.querySelector("div.consumptionMetaDiv > div > h1").textContent`